limits and quotas (#388)
aziontech/docs: src/content/docs/en/pages/main-menu/reference/limits/limits.mdx

<Button href="https://www.azion.com/en/pricing/" text="azion service plans"></Button>

## General limits

Limits are defined by the Azion Edge Platform according to the product being used.

### General

| Scope | Description | Standard Limit |
| ----- | ----------- | ----- |
| Platform | Object in cache | Maximum 10 GB per object |
| Platform | Keep-alive connection | Up to 15 minutes |

### Build

| Scope | Description | Standard Limit |
| ----- | ----------- | ----- |
| [Application Acceleration](/en/documentation/products/edgeapplication/application-acceleration/) | Minimum TTL (custom) | 0 seconds |
| [Domains](/en/documentation/products/edge-application/domains/) | Maximum CNAME per domain | 50 CNAMEs |
| [Edge Application](/en/documentation/products/edge-application/) | Maximum Debug Rules logs | 1500 bytes |
| [Edge Application](/en/documentation/products/edge-application/) | Max number of Criteria per Edge Application Rule Set | 10 criteria |
| [Edge Application](/en/documentation/products/edge-application/) | Max number of Behavior per Edge Application Rule Set | 10 behaviors | 

> | [Edge Caching](/en/documentation/products/edge-application/edge-caching/) | Minimum TTL (default) | 60 seconds |

Author: ```suggestion
| [Edge Caching](/en/documentation/products/edge-application/edge-caching/) | Minimum TTL (default) | 60 seconds |
| [Jamstack Build](/) | Deploy hook triggers per hour | 2 |
| [Jamstack Build](/) | Deployments per hour | 2 |
```

| [L2 Caching](/en/documentation/products/edge-application/l2-caching/) | Content caching | TTL equal or greater than 2592000 seconds (30 days) |
| [Real-Time Purge (History)](/en/documentation/products/edge-application/real-time-purge/) | Maximum number of wildcards URLs from which the Purge was requested | Limited to 6 months and up to 1000000 requests |
| [Real-Time Purge (URL and Cache Key)](/en/documentation/products/edge-application/real-time-purge/) | Operations per month | 1000 operations per month |
| [Real-Time Purge (wildcard)](/en/documentation/products/edge-application/real-time-purge/) | Operations per month | 1000 operations per month |
| [Slice Caching Size](/en/documentation/products/edge-application/cache-settings/#slice-settings) | Slice Caching Size | 1024 kb | 
| Websocket | Connection timeout | Up to 15 minutes |

### Secure

| Scope | Description | Standard Limit |
| ----- | ----------- | ----- |

> | DDoS Protection | Bandwidth traffic | Unmetered |

Author: ```suggestion
| DDoS Protection | Bandwidth traffic | Unmetered |
| Edge Firewall | Network Lists entries | 20 thousand lines |
| Edge Firewall | ASN in a Network List | 20 thousand lines |
| Edge Firewall | IP/CIDR in a Network List | 20 thousand lines |
| Edge Firewall | Web Application Firewall | 128kb |
| Web Application Firewall | Debug Rules logs | 1500 logs |
| Web Application Firewall | WAF Tuning consult queries | 3 days |
```

| [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL maximum payload | 5 GB |
| [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL queries | 120 requests per minute |

---

## Quotas

Quotas are defined by the Azion product being used and the service plan contracted. If a product, module, or feature doesn't have specified quotas, they run on a usage-based approach, meaning you pay as you use them.

<Button href="https://www.azion.com/en/pricing/" text="see products pricing"></Button>

### Build

#### Edge Application

[Edge Application](/en/documentation/products/edge-application/) product, modules, and features usage **quotas** per account, according to each service plan:

| Scope | Quota for Developer Plan | Quota for Business Plan | Quota for Enterprise Plan | Quota for Mission Critical Plan |
| ---- | ---- | ---- | ---- | ---- |
| Maximum number of edge applications | 10 instances | 50 instances | 200 instances | Customizable |
| Maximum number of domains per edge application | 10 instances | 50 instances | 200 instances | Customizable |

> ### Secure
> 
> #### Edge Firewall
> 
> [Edge Firewall](/en/documentation/products/edge-firewall/) product, modules, and features usage **quotas** per account, according to each service plan:
> 
> | Scope | Quota for Developer Plan | Quota for Business Plan | Quota for Enterprise Plan | Quota for Mission Critical Plan |
> | ---- | ---- | ---- | ---- | ---- |
> | Maximum number of edge firewalls | 10 instances | 50 instances | 200 instances | Customizable |
> | Maximum number of domains per edge firewall | All | All | All | All |

Author: ```suggestion
```

| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Serverless Functions Created per Deployment | Framework-dependent* |
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Uncompressed size for a code bundle | 1MB |
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Vercel Projects Connected per Git Repository | 3 |
| [Jamstack Build](/en/documentation/products/build/develop-with-azion/frameworks-specific/overview/) | Deploy hook triggers per hour | 2 |
| [Jamstack Build](/en/documentation/products/build/develop-with-azion/frameworks-specific/overview/) | Deployments per hour | 2 |
| [L2 Caching](/en/documentation/products/edge-application/l2-caching/) | Content caching | TTL equal or greater than 2592000 seconds (30 days) |
| [Real-Time Purge (History)](/en/documentation/products/edge-application/real-time-purge/) | Maximum number of wildcards URLs from which the Purge was requested | Limited to 6 months and up to 1000000 requests |
| [Real-Time Purge (URL and Cache Key)](/en/documentation/products/edge-application/real-time-purge/) | Operations per month | 1000 operations per month |

> | [Real-Time Purge (wildcard)](/en/documentation/products/edge-application/real-time-purge/) | Operations per month | 1000 operations per month |

Reviewer: ```suggestion
| [Real-Time Purge (wildcard)](/en/documentation/products/edge-application/real-time-purge/) | Operations per day (24h interval) | Up to 2000 requests |
| [Real-Time Purge (wildcard)](/en/documentation/products/edge-application/real-time-purge/) | Wildcard URL per request | 1 |
| [Real-Time Purge (wildcard)](/en/documentation/products/edge-application/real-time-purge/) | Size per Wildcard URL  | 256 characters |
```

| [Data Steaming](/en/documentation/products/data-streaming/) | Time for data to be transferred to the client's infrastructure | Up to 3 minutes |
| [Real-Time Events](/en/documentation/products/real-time-events/) | Log retention | 7 days |
| [Real-Time Events](/en/documentation/products/real-time-events/) | Available in | 3 minutes |
| [Real-Time Events](/en/documentation/products/graphql-api-overview/) | GraphQL data transferred | 10000 lines |
| [Real-Time Events](/en/documentation/products/graphql-api-overview/) | GraphQL maximum fields | 10 fields |
| [Real-Time Events](/en/documentation/products/graphql-api-overview/) | GraphQL maximum payload | 5 GB |
| [Real-Time Events](/en/documentation/products/graphql-api-overview/) | GraphQL queries | 120 requests per minute |
| [Real-Time Metrics](/en/documentation/products/real-time-metrics/) | Log retention | 24 months |
| [Real-Time Metrics](/en/documentation/products/real-time-metrics/) | UI queries | 120 requests per minute |
| [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL data transferred | 10000 lines |

> | [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL maximum fields | 35 fields |

Author: fui eu que tinha colocado hahaha acabei de validar com ele e é pra manter!

| [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL maximum payload | 5 GB |
| [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL queries | 120 requests per minute |

---

## Quotas

Quotas are defined by the Azion product being used and the service plan contracted. If a product, module, or feature doesn't have specified quotas, they run on a usage-based approach, meaning you pay as you use them.

<Button href="https://www.azion.com/en/pricing/" text="see products pricing"></Button>

### Build

#### Edge Application

[Edge Application](/en/documentation/products/edge-application/) product, modules, and features usage **quotas** per account, according to each service plan:

| Scope | Quota for Developer Plan | Quota for Business Plan | Quota for Enterprise Plan | Quota for Mission Critical Plan |
| ---- | ---- | ---- | ---- | ---- |
| Maximum number of edge applications | 10 instances | 50 instances | 200 instances | Customizable |

> | Maximum number of domains per edge application | 10 instances | 50 instances | 200 instances | Customizable |

Reviewer: ```suggestion
| Maximum number of domains per edge application | 10 instances | 50 instances | 200 instances | Customizable |
| Maximum purge operations per month | The first 1000 purge operations each month are included for all customers. When exceeding this volume, user will pay for each 1000 additional purges. |
```

Limits are technical and functional capabilities of Azion's architecture and infrastructure.

:::tip
All Azion limits are customizable. If you need to increase the standard availability, contact the [sales team](https://www.azion.com/en/contact-sales/).
:::

Quotas are usability shares defined by the product. These values represent the allocation or distribution of specific resources. Quota values are based on the contracted Service Plan.

<Button href="https://www.azion.com/en/pricing/" text="azion service plans"></Button>

## General limits

Limits are defined by the Azion Edge Platform according to the product being used.

### General

| Scope | Description | Standard Limit |
| ----- | ----------- | ----- |
| Platform | Object in cache | Maximum 10 GB per object |
| Platform | Keep-alive connection | Up to 15 minutes |

### Build

| Scope | Description | Standard Limit |
| ----- | ----------- | ----- |
| [Application Acceleration](/en/documentation/products/edgeapplication/application-acceleration/) | Minimum TTL (custom) | 0 seconds |
| [Domains](/en/documentation/products/edge-application/domains/) | Maximum CNAME per domain | 50 CNAMEs |

> | [Edge Application](/en/documentation/products/edge-application/) | Maximum Debug Rules logs | 1500 bytes |

Author: ```suggestion
| [Edge Application](/en/documentation/products/edge-application/) | Maximum Debug Rules logs | 1,500 bytes |
```

| [Edge Application](/en/documentation/products/edge-application/) | Max number of Criteria per Edge Application Rule Set | 10 criteria |
| [Edge Application](/en/documentation/products/edge-application/) | Max number of Behavior per Edge Application Rule Set | 10 behaviors | 
| [Edge Caching](/en/documentation/products/edge-application/edge-caching/) | Minimum TTL (default) | 60 seconds |

> | [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Arguments  | 100kb |
> | [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Code Limit (API) | 200MB |
> | [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Code Limit (UI) | 5-6MB |

Author: ```suggestion
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Arguments  | 100 kb |
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Code Limit (API) | 200 MB |
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Code Limit (UI) | 5-6 MB |
```

> | [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Environment variable size | 32kb |

Author: ```suggestion
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Environment variable size | 32 kb |
```

| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Maximum number of EdgeWorker IDs per account | 200 |

> | [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Maximum number of versions per EdgeWorker ID | 1000 |

Author: ```suggestion
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Maximum number of versions per EdgeWorker ID | 1,000 |
```

> | [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Memory per Isolate | 512MB |

Author: ```suggestion
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Memory per Isolate | 512 MB |
```

| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Number of EdgeWorker scripts executed per request | 1 |
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Build Time per Deployment (Minutes) | 45 |

> | [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Compressed size for a code bundle | 512KB |

Author: ```suggestion
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Compressed size for a code bundle | 512 KB |
```

| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Concurrent Builds | 1 |

> | [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Deployments Created from CLI per Week | 2000 |
> | [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Disk Size | 13GB |

Author: ```suggestion
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Deployments Created from CLI per Week | 2,000 |
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Disk Size | 13 GB |
```

| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Edge Function Initial Response Timeout | 30 seconds |
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Hostnames included | 100 |

> | [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Max hostnames | 5000 |

Author: ```suggestion
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Max hostnames | 5,000 |
```

| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Maximum body size for responses from an EdgeWorkers function to a client browser | 2048 characters if you use `request.respondWith()` in the `onClientRequest` or `onClientResponse` event handler |
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Maximum body size for responses from origin to an EdgeWorkers function, using `httpRequest()` | 128 KB when using `json()` or `text()` to buffer the entire response. 5 MB when using body to read the response as a stream |
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Maximum body size when `request.text()` or `request.json()` is called in the `responseProvider` event handler | 16KB |
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Maximum number of activations each day per account | 500 |
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Maximum number of activations each minute on a network per account | 20 |

> | [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Maximum response header size for HTTP sub-requests | 8000 bytes |

Author: ```suggestion
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Maximum response header size for HTTP sub-requests | 8,000 bytes |
```

| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Maximum size of the response status and header for responses generated using createResponse | 8KB |

> | [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Maximum streaming request body size when `request.body` is called in the `responseProvider` event handler | 1MB |

Author: ```suggestion
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Maximum streaming request body size when `request.body` is called in the `responseProvider` event handler | 1 MB |
```

| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Number of Cache rules | 10 |
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Proxied Request Timeout | 30 seconds |
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Serverless Function Execution Timeout | 10 seconds |
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Serverless Functions Created per Deployment | Framework-dependent* |

> | [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Uncompressed size for a code bundle | 1MB |

Author: ```suggestion
| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Uncompressed size for a code bundle | 1 MB |
```

| [Edge Functions](/en/documentation/products/edge-application/edge-functions/) | Vercel Projects Connected per Git Repository | 3 |
| [Jamstack Build](/en/documentation/products/build/develop-with-azion/frameworks-specific/overview/) | Deploy hook triggers per hour | 2 |
| [Jamstack Build](/en/documentation/products/build/develop-with-azion/frameworks-specific/overview/) | Deployments per hour | 2 |

> | [L2 Caching](/en/documentation/products/edge-application/l2-caching/) | Content caching | TTL equal or greater than 2592000 seconds (30 days) |

Author: ```suggestion
| [L2 Caching](/en/documentation/products/edge-application/l2-caching/) | Content caching | TTL equal or greater than 2,592,000 seconds (30 days) |
```

> | [Real-Time Purge (History)](/en/documentation/products/edge-application/real-time-purge/) | Maximum number of wildcards URLs from which the Purge was requested | Limited to 6 months and up to 1000000 requests |

Author: ```suggestion
| [Real-Time Purge (History)](/en/documentation/products/edge-application/real-time-purge/) | Maximum number of wildcards URLs from which the Purge was requested | Limited to 6 months and up to 1,000,000 requests |
```

| [Real-Time Purge (URL and Cache Key)](/en/documentation/products/edge-application/real-time-purge/) | Operations per minute | Up to 10,000 objects |
| [Real-Time Purge (URL and Cache Key)](/en/documentation/products/edge-application/real-time-purge/) | Operations per client | Up to 200 requests with 50 objects per request |

> | [Real-Time Purge (URL and Cache Key)](/en/documentation/products/edge-application/real-time-purge/) | Size per URL or Cache Key | 4096 characters |

Author: ```suggestion
| [Real-Time Purge (URL and Cache Key)](/en/documentation/products/edge-application/real-time-purge/) | Size per URL or Cache Key | 4,096 characters |
```

| [Real-Time Purge (wildcard)](/en/documentation/products/edge-application/real-time-purge/) | Operations per day (24h interval) | Up to 2,000 requests |
| [Real-Time Purge (wildcard)](/en/documentation/products/edge-application/real-time-purge/) | Wildcard URL per request | 1 |

> | [Real-Time Purge (wildcard)](/en/documentation/products/edge-application/real-time-purge/) | Size per Wildcard URL  | 256 characters |

Author: ```suggestion
| [Real-Time Purge (wildcard)](/en/documentation/products/edge-application/real-time-purge/) | Size per Wildcard URL | 256 characters |
```

> | [Slice Caching Size](/en/documentation/products/edge-application/cache-settings/#slice-settings) | Slice Caching Size | 1024 kb | 

Author: ```suggestion
| [Slice Caching Size](/en/documentation/products/edge-application/cache-settings/#slice-settings) | Slice Caching Size | 1,024 kb | 
```

| Websocket | Connection timeout | Up to 15 minutes |

### Secure

| Scope | Description | Standard Limit |
| ----- | ----------- | ----- |
| DDoS Protection | Bandwidth traffic | Unmetered |
| Edge Firewall | Network Lists entries | 20,000 lines |
| Edge Firewall | ASN in a Network List | 20,000 lines |
| Edge Firewall | IP/CIDR in a Network List | 20,000 lines |

> | Edge Firewall | Web Application Firewall | 128kb |

Author: ```suggestion
| Edge Firewall | Web Application Firewall | 128 kb |
```

| Web Application Firewall | Maximum Debug Rules logs | 1,500 bytes |
| Web Application Firewall | WAF Tuning consult queries | 3 days |

### Observe

| Scope | Description | Standard Limit |
| ----- | ----------- | ----- |

> | [Data Streaming](/en/documentation/products/data-streaming/) | Dispatch | 60 seconds or blocks of 2000 requests, whichever happens first. For the AWS Kinesis Data Firehose endpoint: 60 seconds or blocks of 500 requests |

Author: ```suggestion
| [Data Streaming](/en/documentation/products/data-streaming/) | Dispatch | 60 seconds or blocks of 2,000 requests, whichever happens first. For the AWS Kinesis Data Firehose endpoint: 60 seconds or blocks of 500 requests |
```

| [Data Streaming](/en/documentation/products/data-streaming/) | Delay | Lower than 3 minutes |
| [Data Streaming](/en/documentation/products/data-streaming/) | Sampling | 1 active stream |

> | [Data Streaming](/en/documentation/products/data-streaming/) | UI access | Users with more than 3000 domains must use the API to edit streams |

Author: ```suggestion
| [Data Streaming](/en/documentation/products/data-streaming/) | UI access | Users with more than 3,000 domains must use the API to edit streams |
```

| [Data Streaming](/en/documentation/products/data-streaming/) | Time for data to be transferred to the client's infrastructure | Up to 3 minutes |
| [Real-Time Events](/en/documentation/products/real-time-events/) | Log retention | 7 days |
| [Real-Time Events](/en/documentation/products/real-time-events/) | Available in | 3 minutes |

> | [Real-Time Events](/en/documentation/products/graphql-api-overview/) | GraphQL data transferred | 10000 lines |

Author: ```suggestion
| [Real-Time Events](/en/documentation/products/graphql-api-overview/) | GraphQL data transferred | 10,000 lines |
```

| [Real-Time Events](/en/documentation/products/graphql-api-overview/) | GraphQL maximum fields | 10 fields |
| [Real-Time Events](/en/documentation/products/graphql-api-overview/) | GraphQL maximum payload | 5 GB |
| [Real-Time Events](/en/documentation/products/graphql-api-overview/) | GraphQL queries | 120 requests per minute |
| [Real-Time Metrics](/en/documentation/products/real-time-metrics/) | Log retention | 24 months |
| [Real-Time Metrics](/en/documentation/products/real-time-metrics/) | UI queries | 120 requests per minute |

> | [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL data transferred | 10000 lines |

Author: ```suggestion
| [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL data transferred | 10,000 lines |
```

---
title: Limits and quotas
description: Get to know the limits and quotas available for each Azion Product and Service Plan.
meta_tags: azion, edge computing, limits, quotas
namespace: docs_limits_quotas
permalink: /documentation/limits-and-quotas/
---

import Button from '~/components/Button.astro'

This reference documentation covers the user **limits** for Azion Products and their respective **quotas** for each **Service Plan**.

> 

Reviewer: :::tip
```suggestion
All Azion limits and quotas are customizable. If you need to increase the standard availability, contact the [sales team](https://www.azion.com/en/contact-sales/).
```
:::

Limits are technical and functional capabilities of Azion's architecture and infrastructure.

> :::tip

Reviewer: ```suggestion

```

> All Azion limits are customizable. If you need to increase the standard availability, contact the [sales team](https://www.azion.com/en/contact-sales/).

Reviewer: ```suggestion

```

> :::

Reviewer: ```suggestion

```

Quotas are usability shares defined by the product. These values represent the allocation or distribution of specific resources. Quota values are based on the contracted Service Plan.

<Button href="https://www.azion.com/en/pricing/" text="azion service plans"></Button>

## General limits

Limits are defined by the Azion Edge Platform according to the product being used.

### General

| Scope | Description | Standard Limit |
| ----- | ----------- | ----- |

> | Platform | Object in cache | Maximum 10 GB per object |

Author: ```suggestion
| [Platform](/en/documentation/agreements/tos/#1-products-and-services) | Object in cache | Maximum 10 GB per object |
```

| [Slice Caching Size](/en/documentation/products/edge-application/cache-settings/#slice-settings) | Slice Caching Size | 1,024 kb | 
| Websocket | Connection timeout | Up to 15 minutes |

### Secure

| Scope | Description | Standard Limit |
| ----- | ----------- | ----- |
| [DDoS Protection](/en/documentation/products/edge-firewall/ddos-protection/) | Bandwidth traffic | Unmetered |
| [Edge Firewall](/en/documentation/products/edge-firewall/) | Network Lists entries | 20,000 lines |
| [Edge Firewall](/en/documentation/products/edge-firewall/) | ASN in a Network List | 20,000 lines |
| [Edge Firewall](/en/documentation/products/edge-firewall/) | IP/CIDR in a Network List | 20,000 lines |
| [Edge Firewall](/en/documentation/products/edge-firewall/) | Web Application Firewall | 128 kb |
| [Web Application Firewall](/en/documentation/products/edge-firewall/web-application-firewall/) | Maximum Debug Rules logs | 1,500 bytes |
| [Web Application Firewall](/en/documentation/products/edge-firewall/web-application-firewall/) | WAF Tuning consult queries | 3 days |

### Observe

| Scope | Description | Standard Limit |
| ----- | ----------- | ----- |

> | [Data Streaming](/en/documentation/products/data-streaming/) | Dispatch | 60 seconds or blocks of 2,000 requests, whichever happens first. For the AWS Kinesis Data Firehose endpoint: 60 seconds or blocks of 500 requests |

Author: ```suggestion
| [Data Streaming](/en/documentation/products/data-streaming/) | Dispatch | 60 seconds or packages of 2,000 requests, whichever happens first. For the AWS Kinesis Data Firehose endpoint: 60 seconds or packages of 500 requests |
```

| [Real-Time Events](/en/documentation/products/graphql-api-overview/) | GraphQL queries | 120 requests per minute |
| [Real-Time Metrics](/en/documentation/products/real-time-metrics/) | Log retention | 24 months |
| [Real-Time Metrics](/en/documentation/products/real-time-metrics/) | UI queries | 120 requests per minute |
| [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL data transferred | 10,000 lines |
| [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL maximum fields | 35 fields |
| [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL maximum payload | 5 GB |
| [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL queries | 120 requests per minute |

---

## Quotas

Quotas are defined by the Azion product being used and the service plan contracted. If a product, module, or feature doesn't have specified quotas, they run on a usage-based approach, meaning you pay as you use them.

<Button href="https://www.azion.com/en/pricing/" text="see products pricing"></Button>

### Build

#### Edge Application

> [Edge Application](/en/documentation/products/edge-application/) product, modules, and features usage **quotas** per account, according to each service plan:

Author: ```suggestion
[Edge Application](/en/documentation/products/edge-application/) product, modules, and features usage **quotas** per account, according to each [service plan](https://www.azion.com/en/pricing/):
```

> | Scope | Quota for Developer Plan | Quota for Business Plan | Quota for Enterprise Plan | Quota for Mission Critical Plan |

Author: ```suggestion
| Scope | Developer | Busines | Enterprise | Mission Critical |
```

description: Get to know the limits and quotas available for each Azion Product and Service Plan.
meta_tags: azion, edge computing, limits, quotas
namespace: docs_limits_quotas
permalink: /documentation/limits-and-quotas/
---

import Button from '~/components/Button.astro'

This reference documentation covers the user **limits** for Azion Products and their respective **quotas** for each **Service Plan**.

Limits are technical and functional capabilities of Azion's architecture and infrastructure.

:::tip
All Azion limits are customizable. If you need to increase the standard availability, contact the [sales team](https://www.azion.com/en/contact-sales/).
:::

Quotas are usability shares defined by the product. These values represent the allocation or distribution of specific resources. Quota values are based on the contracted Service Plan.

<Button href="https://www.azion.com/en/pricing/" text="azion service plans"></Button>

## General limits

Limits are defined by the Azion Edge Platform according to the product being used.

### General

| Scope | Description | Standard Limit |
| ----- | ----------- | ----- |
| Platform | Object in cache | Maximum 10 GB per object |

> | Platform | Keep-alive connection | Up to 15 minutes |

Author: ```suggestion
| [Platform](/en/documentation/products/edge-application/#limits) | Keep-alive connection | Up to 15 minutes |
```

| [Real-Time Purge (URL and Cache Key)](/en/documentation/products/edge-application/real-time-purge/) | Operations per minute | Up to 10,000 objects |
| [Real-Time Purge (URL and Cache Key)](/en/documentation/products/edge-application/real-time-purge/) | Operations per client | Up to 200 requests with 50 objects per request |
| [Real-Time Purge (URL and Cache Key)](/en/documentation/products/edge-application/real-time-purge/) | Size per URL or Cache Key | 4,096 characters |
| [Real-Time Purge (wildcard)](/en/documentation/products/edge-application/real-time-purge/) | Operations per day (24h interval) | Up to 2,000 requests |
| [Real-Time Purge (wildcard)](/en/documentation/products/edge-application/real-time-purge/) | Wildcard URL per request | 1 |
| [Real-Time Purge (wildcard)](/en/documentation/products/edge-application/real-time-purge/) | Size per Wildcard URL | 256 characters |
| [Slice Caching Size](/en/documentation/products/edge-application/cache-settings/#slice-settings) | Slice Caching Size | 1,024 kb | 
| Websocket | Connection timeout | Up to 15 minutes |

### Secure

| Scope | Description | Standard Limit |
| ----- | ----------- | ----- |

> | DDoS Protection | Bandwidth traffic | Unmetered |

Author: ```suggestion
| [DDoS Protection](/en/documentation/products/edge-firewall/ddos-protection/) | Bandwidth traffic | Unmetered |
```

> | Edge Firewall | Network Lists entries | 20,000 lines |
> | Edge Firewall | ASN in a Network List | 20,000 lines |
> | Edge Firewall | IP/CIDR in a Network List | 20,000 lines |
> | Edge Firewall | Web Application Firewall | 128 kb |
> | Web Application Firewall | Maximum Debug Rules logs | 1,500 bytes |
> | Web Application Firewall | WAF Tuning consult queries | 3 days |

Author: ```suggestion
| [Edge Firewall](/en/documentation/products/edge-firewall/) | Network Lists entries | 20,000 lines |
| [Edge Firewall](/en/documentation/products/edge-firewall/) | ASN in a Network List | 20,000 lines |
| [Edge Firewall](/en/documentation/products/edge-firewall/) | IP/CIDR in a Network List | 20,000 lines |
| [Edge Firewall](/en/documentation/products/edge-firewall/) | Web Application Firewall | 128 kb |
| [Web Application Firewall](/en/documentation/products/edge-firewall/web-application-firewall/) | Maximum Debug Rules logs | 1,500 bytes |
| [Web Application Firewall](/en/documentation/products/edge-firewall/web-application-firewall/) | WAF Tuning consult queries | 3 days |
```

| [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL maximum payload | 5 GB |
| [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL queries | 120 requests per minute |

---

## Quotas

Quotas are defined by the Azion product being used and the service plan contracted. If a product, module, or feature doesn't have specified quotas, they run on a usage-based approach, meaning you pay as you use them.

<Button href="https://www.azion.com/en/pricing/" text="see products pricing"></Button>

### Build

#### Edge Application

[Edge Application](/en/documentation/products/edge-application/) product, modules, and features usage **quotas** per account, according to each service plan:

| Scope | Quota for Developer Plan | Quota for Business Plan | Quota for Enterprise Plan | Quota for Mission Critical Plan |
| ---- | ---- | ---- | ---- | ---- |
| Maximum number of edge applications | 10 instances | 50 instances | 200 instances | Customizable |
| Maximum number of domains per edge application | 10 instances | 50 instances | 200 instances | Customizable |
| Maximum purge operations per month | 1,000 | 1,000* | 1,000* | 1,000* |

> * The first 1000 purge operations per each month are included for all customers. After exceeding this volume, users will pay for each 1,000 additional purges.

Author: ```suggestion
*\ The first 1000 purge operations per each month are included for all customers. After exceeding this volume, users will pay for each 1,000 additional purges.
```

---
title: Limits and quotas
description: Get to know the limits and quotas available for each Azion Product and Service Plan.
meta_tags: azion, edge computing, limits, quotas
namespace: docs_limits_quotas
permalink: /documentation/limits-and-quotas/
---

import Button from '~/components/Button.astro'

This reference documentation covers the user **limits** for Azion Products and their respective **quotas** for each **Service Plan**.

Limits are technical and functional capabilities of Azion's architecture and infrastructure.

:::tip
All Azion limits are customizable. If you need to increase the standard availability, contact the [support team](/en/documentation/services/support/#service-channels).
:::

Quotas are usability shares defined by the product. These values represent the allocation or distribution of specific resources. Quota values are based on the contracted Service Plan.

<Button href="https://www.azion.com/en/pricing/" text="azion service plans"></Button>

## General limits

Limits are defined by the Azion Edge Platform according to the product being used.

### General

> | Scope | Description | Limit |

Reviewer: ```suggestion
| Scope | Description | Standard Limit |
```

Reviewer: Aqui a ideia de colocar o "Standard" é mostrar que esse é o limite básico, para reforçar que pode ser customizado.

| ----- | ----------- | ----- |
| Platform | Object in cache | Maximum 10 GB per object |
| Platform | Keep-alive connection | Up to 15 minutes |

### Build

> | Scope | Description | Limit |

Reviewer: ```suggestion
| Scope | Description | Standard Limit |
```

| ----- | ----------- | ----- |
| [Application Acceleration](/en/documentation/products/edgeapplication/application-acceleration/) | Minimum TTL (custom) | 0 seconds |
| [Domains](/en/documentation/products/edge-application/domains/) | Maximum CNAME per domain | 50 CNAMEs |
| [Edge Application](/en/documentation/products/edge-application/) | Maximum Debug Rules logs | 1500 bytes |
| [Edge Application](/en/documentation/products/edge-application/) | Max number of Criteria per Edge Application Rule Set | 10 criteria |
| [Edge Application](/en/documentation/products/edge-application/) | Max number of Behavior per Edge Application Rule Set | 10 behaviors | 
| [Edge Caching](/en/documentation/products/edge-application/edge-caching/) | Minimum TTL (default) | 60 seconds |

> | [L2 Caching](/en/documentation/products/edge-application/l2-caching/) | Content cahing | TTL equal or greater than 2592000 seconds (30 days) |

Reviewer: ```suggestion
| [L2 Caching](/en/documentation/products/edge-application/l2-caching/) | Content caching | TTL equal or greater than 2592000 seconds (30 days) |
```

> | [Real-Time Purge (History)](/en/documentation/products/edge-application/real-time-purge/) | Maximum number of wildcards URLs from which the Purge was requested. | Limited to 6 months and up to 1000000 requests |

Reviewer: ```suggestion
| [Real-Time Purge (History)](/en/documentation/products/edge-application/real-time-purge/) | Maximum number of wildcards URLs from which the Purge was requested | Limited to 6 months and up to 1000000 requests |
```

| [Real-Time Purge (URL and Cache Key)](/en/documentation/products/edge-application/real-time-purge/) | Operations per month | 1000 operations per month |
| [Real-Time Purge (wildcard)](/en/documentation/products/edge-application/real-time-purge/) | Operations per month | 1000 operations per month |
| [Slice Caching Size](/en/documentation/products/edge-application/cache-settings/#slice-settings) | Slice Caching Size | 1024 kb | 
| Websocket | Connection timeout | Up to 15 minutes |

### Secure

> | Scope | Description | Limit |

Author: ```suggestion
| Scope | Description | Standard Limit |
```

| ----- | ----------- | ----- |

> |  |  |

Author: ```suggestion
| DDoS Protection | DDoS Protection - 20 Gbps | Unmetered |
| DDoS Protection | DDoS Protection - 5 Gbps | Unmetered |
| DDoS Protection | DDoS Protection - 50 Gbps | Unmetered |
| DDoS Protection | DDoS Protection - Unlimited | Unmetered |
```

### Observe

> | Scope | Description | Limit |

Author: ```suggestion
| Scope | Description | Standard Limit |
```

| [Real-Time Events](/en/documentation/products/graphql-api-overview/) | GraphQL data transferred | 10000 lines |
| [Real-Time Events](/en/documentation/products/graphql-api-overview/) | GraphQL maximum fields | 10 fields |
| [Real-Time Events](/en/documentation/products/graphql-api-overview/) | GraphQL maximum payload | 5 GB |
| [Real-Time Events](/en/documentation/products/graphql-api-overview/) | GraphQL queries | 120 requests per minute |
| [Real-Time Metrics](/en/documentation/products/real-time-metrics/) | Log retention | 24 months |
| [Real-Time Metrics](/en/documentation/products/real-time-metrics/) | UI queries | 120 requests per minute |
| [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL data transferred | 10000 lines |
| [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL maximum fields | 35 fields |
| [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL maximum payload | 5 GB |
| [Real-Time Metrics](/en/documentation/products/graphql-api-overview/) | GraphQL queries | 120 requests per minute |

---

## Quotas

> Quotas are defined by the Azion product being used and the service plan contracted. If a product, module, or feature doesn't have specified quotas, they run on a usage-based approach, meaning you pay as you use them.

Reviewer: ```suggestion
Quotas are defined by the Azion product being used and the service plan contracted. If a product, module or feature doesn't have specified quotas, they run on a usage-based approach, meaning you pay as you use them.
```

<Button href="https://www.azion.com/en/pricing/" text="see products pricing"></Button>

### Build

#### Edge Application

> [Edge Application](/en/documentation/products/edge-application/) product, modules, and features usage **quotas**, according to each service plan:

Reviewer: ```suggestion
[Edge Application](/en/documentation/products/edge-application/) product, modules, and features usage **quotas** per account, according to each service plan:
```

| Scope | Quota for Developer Plan | Quota for Business Plan | Quota for Enterprise Plan | Quota for Mission Critical Plan |
| ---- | ---- | ---- | ---- | ---- |
| Maximum number of edge applications | 10 instances | 50 instances | 200 instances | Customizable |

> | Maximum number of domains per edge application | 10 instances | 50 instances | 200 instances | Customizable |

Author: ```suggestion
| Maximum number of domains per edge application | 10 instances | 50 instances | 200 instances | Customizable |

### Secure

#### Edge Firewall

[Edge Firewall](/en/documentation/products/edge-firewall/) product, modules, and features usage **quotas** per account, according to each service plan:

| Scope | Quota for Developer Plan | Quota for Business Plan | Quota for Enterprise Plan | Quota for Mission Critical Plan |
| ---- | ---- | ---- | ---- | ---- |
| Maximum number of edge firewalls | 10 instances | 50 instances | 200 instances | Customizable |
| Maximum number of domains per edge firewall | All | All | All | All |

```
